src/main/java/com/eternalcode/parcellockers/database/ParcelLockerDatabaseService.java

```diff
@@ -0,0 +1,222 @@
+                 PreparedStatement statement = connection.prepareStatement(
+                         "SELECT * FROM `parcellockers`;"
+                 )
+            ) {
+                return this.extractParcelLockers(statement);
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+
+    @Override
+    public CompletableFuture<Optional<ParcelLocker>> findByUUID(UUID uuid) {
+        return CompletableFuture.supplyAsync(() -> {
+            try (Connection connection = this.dataSource.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                         "SELECT * FROM `parcellockers` WHERE `uuid` = ?;"
+                 )
+            ) {
+                statement.setString(1, uuid.toString());
+                ResultSet rs = statement.executeQuery();
+                ParcelLocker parcelLocker = null;
+
+                if (rs.next()) {
+                    parcelLocker = new ParcelLocker(
+                            UUID.fromString(rs.getString("uuid")),
+                            rs.getString("description"),
+                            Position.parse(rs.getString("position"))
+                    );
+                }
+                this.cache.put(parcelLocker.uuid(), parcelLocker);
+                return Optional.ofNullable(parcelLocker);
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public CompletableFuture<Optional<ParcelLocker>> findByPosition(Position position) {
+        return CompletableFuture.supplyAsync(() -> {
+            try (Connection connection = this.dataSource.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                         "SELECT * FROM `parcelLockers` WHERE `position` = ?;"
+                 )
+            ) {
+                statement.setString(1, position.toString());
+                ResultSet rs = statement.executeQuery();
+                ParcelLocker parcelLocker = null;
+
+                if (rs.next()) {
+                    parcelLocker = new ParcelLocker(
+                            UUID.fromString(rs.getString("uuid")),
+                            rs.getString("description"),
+                            Position.parse(rs.getString("position"))
+                    );
+                }
+                this.cache.put(parcelLocker.uuid(), parcelLocker);
+                return Optional.ofNullable(parcelLocker);
+            }
+            catch (SQLException e) {
+                Sentry.captureException(e);
+                throw new ParcelLockersException(e);
+            }
+        }).orTimeout(5, TimeUnit.SECONDS);
+    }
```

Comment:
zrób może sobie jakąś prywatną metodę, która szuka tego parcel lockera i w argumencie metody dajesz string i tam, by się wpisywało np "position", wtedy dodajesz do statementa tego stringa i masz kod 2 razy krótszy. Chodzi mi o to, że później w tych metodach findByPosition itp używasz tej gotowej metody tylko w tym argumencie podajesz po czym ma szukać

Reply:
Mogę prosić o example? Próbowałem zrobić ekstrakcję kodu na podstawie duplikatów, ale w jednej z metod nie mogę podać parametru `position`